### PR TITLE
Add opt-in default for "Always open List View" in the Block Editor (#32)

### LINKED ISSUE #32: Please let the browser extension default "Always open List View" in the Block Editor to on for all WordPress sites.
**What problem does this solve?**

Can the browser extension please have an option to automatically enable the "Always open List View" setting in the block editor on every WordPress site I visit?  Having to go and enable that on each new site is the itchy tag on my tee shirt of my existence.

**Describe the feature:**

**Acceptance criteria** (what does "done" look like — what should the user be able to do?):

- It somehow sets localstorage or something when visiting a wp site so that list view will default to on.
-
-

**Alternatives considered:**

crying myself to sleep

**Permissions implications**

### PR DESCRIPTION
**Pull request by roshniahuja**
## Summary
Closes #32. Adds an opt-in, off-by-default option that turns on "Always open
List View" for the current user across every WordPress site's Block Editor
(post editor + Site Editor) — a one-time toggle instead of the per-site
manual step the issue describes.

## How it works
- WordPress persists this preference server-side under a single shared
  `"core"` preferences scope (`getPreference("core", "showListViewByDefault")`
  — same call in both edit-post.js and edit-site.js), not purely
  localStorage as the issue's acceptance criteria suggested. localStorage
  is still involved as a local write-through cache with timestamp-based
  conflict resolution, but the source of truth is server-side user meta,
  and both are read/written together through one preferences store.
- `lib/editor-preferences.js` reads WordPress's own already-inlined
  preferences snapshot from the DOM (the same data `wp.data` hydrates from)
  to check, with no REST call, whether the preference is already on.
- If it isn't, content.js asks background.js to dispatch
  `wp.data.dispatch('core/preferences').set('core', 'showListViewByDefault', true)`
  in the page's MAIN world — the same store action a user's own click on
  that Editor Preferences toggle triggers. WordPress's own existing
  debounced local-cache + REST persistence handles the actual save, so the
  extension never touches the REST API or the stored JSON directly — no
  risk of clobbering other saved preferences (dark mode, panel state, etc).

## Data & privacy / permissions
- No new `manifest.json` permissions — `storage` and `scripting` were
  already granted; the write happens through the page's own
  already-authenticated `wp.data`, not a new endpoint the extension calls.
- Off by default. Only ever turns the preference on, never off.
- New global option lives alongside the existing admin-bar/site-info
  defaults in `wp_preferences_v1._global` — no new storage keys.

## Open question for maintainers
The issue's phrasing implies "just make it always on." I made it an
explicit opt-in toggle instead, matching the Site Info panel / admin-bar
precedent in this repo, since it writes to the signed-in user's own
account preferences. Happy to flip it to on-by-default if you'd rather
match the issue literally.

## Test plan
- [x] `cd test && npm test` — new smoke coverage for
  `lib/editor-preferences.js` (DOM snapshot parsing, the "needs it" gate,
  new-account edge case, missing-preference-key edge case)
- [x] `npm run build` — clean
- [x] Manually verified end-to-end against a real logged-in WordPress
  editor session: baseline `showListViewByDefault: false` → enable the new
  option → reload the editor → `true`, confirmed persisted server-side via
  a fresh page load re-reading WordPress's own hydrated snapshot (not
  leftover client state) → reload again, stable, no repeat writes, no
  console errors

### CODE REVIEW

File: lib/editor-preferences.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+   * True unless `serverData` already has the preference set to `true` under
+   * the shared "core" scope. Missing data (null/not-yet-persisted) is
+   * treated as "needs it" — safe because the actual write is a `wp.data`
+   * dispatch that only turns the preference on, never off; a false positive
+   * here at worst repeats a no-op set, not a wrong value.
+   */
+  function needsListViewDefault(serverData) {
+    if (!serverData) return true;
+    return (serverData[SCOPE] || {})[PREFERENCE_NAME] !== true;
+  }
```
Comment (itzmekhokan) [inline lib/editor-preferences.js:70]: This fights a user who turns List View off. This is where "not true → needs it" originates.
- If adding the "assert once per origin" guard:
  - `content.js` — inside the `loadListViewDefaultPref().then(...)` block (around the `chrome.runtime.sendMessage({ type: 'ENSURE_LIST_VIEW_DEFAULT' }) call)`: check/skip on the per-origin marker.
  - `background.js` — in `ensureListViewDefault()`: read the `chrome.scripting.executeScript return ([{ result }], currently discarded)` and only persist the marker when `result === true`.

File: content.js [hunk 2]
```diff
@@ -162,6 +154,34 @@
...
   }
 
+  // -- List View default (Block Editor) -------------------------------------
+  // Opt-in (#32): turns on "Always open List View" for the current user,
+  // mirroring what clicking that toggle in Editor Preferences does. See
+  // lib/editor-preferences.js for why this reads the current value from the
+  // DOM instead of a REST call, and background.js for the MAIN-world
+  // dispatch that actually flips it (content scripts can't reach the page's
+  // `window.wp`).
+  const isBlockEditorPage = isWpAdmin && document.body.classList.contains('block-editor-page');
```
Comment (itzmekhokan) [inline content.js:187]: @roshniahuja lets confirm the class is present on site-editor.php at document_idle; adjust the selector if not.

File: background.js [hunk 3]
```diff
@@ -19,6 +19,11 @@ importScripts('lib/my-sites.js');
 // admin-URL priority chain and its same-origin guard.
 importScripts('lib/rest.js');
 
+// Editor-preferences constants (globalThis.WPEditorPreferences), shared with
+// content.js so the scope list can't drift between the read-side check there
+// and the write-side dispatch here.
+importScripts('lib/editor-preferences.js');
```
Comment (itzmekhokan) [inline background.js:25]: @roshniahuja 
- This is the line that will actually throw in the Safari service worker, because `lib/editor-preferences.js` doesn't exist under `safari/.../Resources/lib/`. Comment: "This new file isn't mirrored into the Safari Resources/ — the SW importScripts will fail. Run `npm run build:safari` and commit."
- Alternate anchor: `manifest.json`, on the added "lib/editor-preferences.js" entry in the content_scripts array — same point, from the content-script side.

Either works; I'd put it on the `background.js` importScripts line since that's the hard failure. No new source file or comment is needed — the fix is running `npm run build:safari` and committing the resulting safari/ changes.

**Review by itzmekhokan — CHANGES_REQUESTED** [review]

File: lib/editor-preferences.js (hunk 1 above)
Comment (roshniahuja) [reply lib/editor-preferences.js:70]: Fixed. Added a one-shot per-origin marker (`listViewDefaultApplied`): `background.js` captures the actual dispatch result and only records the marker on confirmed success, `content.js` skips the check once it's set — so it nudges the pref on exactly once per site and a later manual turn-off sticks. After rebasing onto #58 I routed that write through the new `mutatePref`/`enqueueStorageWrite` serialization instead of a raw read-modify-write. Verified live: enable → on; manually turn off → reload → stays off.

File: content.js (hunk 2 above)
Comment (roshniahuja) [reply content.js:187]: Confirmed on `site-editor.php` at `document_idle` — it's actually server-rendered into the initial HTML (`admin-header.php` sets `block-editor-page` when `is_block_editor()` is true), so it's present before any JS runs, on both the post editor and the Site Editor. Selector unchanged.

File: background.js (hunk 3 above)
Comment (roshniahuja) [reply background.js:25]: Good catch. Added `lib/editor-preferences.js` to `sync-safari.sh` and re-synced the Resources mirror. Also did a full Xcode build to confirm: the file is bundled into the built `.appex` and both the bundled manifest and `background.js` reference it, so the service worker no longer throws.
